File: Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
                res=[]
        if root == None:
            return res
                        q1 = []
        q1.append(root)
                while( len(q1) >0):
            currentlevel=[]
            for i in range(len(q1)):                
                node = q1.pop(0)
                currentlevel.append(node.val)
                if (node.left):
                    q1.append(node.left)
                if (node.right):
                    q1.append(node.right)
                        res.append(currentlevel)
        return res     
                                                                                

File: Binary Tree Level Order Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
vector<vector<int>> sol;
vector<int> lvl;
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        sol.clear();
        if (root == NULL)  return sol;
            queue<TreeNode *> q;
            q.push(root);
            lvl.push_back(root->val);
            while(q.empty() == false) {
                TreeNode* temp = q.front();
                if (lvl.size()!= 0 && lvl.size() == q.size()){
                sol.push_back(lvl);
                                lvl.clear();
                }
                q.pop();
                                if (temp->left != NULL) {
                    q.push(temp->left);
                    lvl.push_back(temp->left->val);
                }
                if (temp->right != NULL){
                    q.push(temp->right);
                    lvl.push_back(temp->right->val);
                }
                            }
        //reverse(sol.begin(),sol.end());
        return sol;
            }
};